src/logic/calculate.js

```diff
@@ -0,0 +1,105 @@
+import operate from './operate';
+
+const calculate = (dataObject, buttonName) => {
+  let { total, next, operation } = dataObject;
+
+  const ArithmeticOperators = ['+', '-', 'X', '÷'];
+
+  const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.'];
+  const modifierOrPercent = ['+/-', '%'];
+  const modifierOrPercentFunc = (number, operator) => {
+    if (next) {
+      if (operation && !total) {
+        operation = null;
+      }
+      next = operate(next, number, operator);
+    } else if (total) {
+      operation = null;
+      total = operate(total, number, operator);
+    }
+  };
+
+  if (buttonName === 'AC') {
+    total = null;
+    next = null;
+    operation = null;
+  } else if (modifierOrPercent.includes(buttonName)) {
+    if (buttonName === '+/-') {
+      modifierOrPercentFunc(-1, 'X');
+    } else if (buttonName === '%') {
+      modifierOrPercentFunc(100, '÷');
+    }
+  } else if (buttonName === '=') {
+    if (total && next && operation) {
+      total = operate(total, next, operation);
+      next = null;
+      operation = null;
+    }
+  } else if (ArithmeticOperators.includes(buttonName)) {
+    if (total && next && operation) {
+      total = operate(total, next, operation);
+      next = null;
+      operation = buttonName;
+    } else {
+      operation = buttonName;
+    }
+  } else if (numbers.includes(buttonName)) {
```

Comment:
You don't need to check for numbers if you have checked for all other possible values of `buttonName`

Reply:
I actually did that because I was getting a linter error `Unexpected if as the only statement in an else block`. I'll check for a `buttonName` whose block will not contain an if statement and use it as my `else` block instead.